understanding-jwt/README.md

```diff
@@ -0,0 +1,206 @@
+
+## Introduction: What, Where and Why?
+
+JWT is a way for securely transmitting information across the network as a JSON object. This information can be verified and trusted because it is digitally signed.
+
+> Well, what does JWT stand for? 
+
+JWT is the abbreviation for JSON Web Token. 
+
+> Now what is JSON? 
+
+That is secret. Actually no. In the spirit of 'Learn by Doing', it will be awesome if you do a bit of research on it if you are hearing it for the first time.
+
+> What does it look like?
+
+`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJDcmlvLmRvIGhlbHBzIHlvdSBsZWFybiBieSBkb2luZyEifQ.QKVJvVR80IMeoRzcLPWh6AIceQs-HkNlMey4swkHOX0`
+
+Intimidating? Let us try to break it down. If you look closely, it contains three parts separated by two `.`.
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.
+eyJzdWIiOiJDcmlvLmRvIGhlbHBzIHlvdSBsZWFybiBieSBkb2luZyEifQ.
+QKVJvVR80IMeoRzcLPWh6AIceQs-HkNlMey4swkHOX0
+```
+Now it is a bit more precise.
+
+> But where is the JSON? This looks like random characters and not JSON.
+
+Yes, you are right; this is not a standard JSON at first sight. This is base64url encoded JSON. base64url is a way to convert a human-readable string into URL friendly string. So, what you saw above is a JSON itself, just a different representation.
+
+The three parts of JWT are
+- ** Header**: The Header is used to give the information to the parser about the token itself. The Header typically consists of the type of the token, which is JWT, and the signing algorithm being used. 
+- ** Payload**: The second part of the token is the Payload, which contains the actual information 
+- ** Signature**: Signature is used to verify the token was not changed along the way. This prevents tampering.
+
+> Where is JWT used?
+
+It is used in almost all web applications to authorize any request to the server. It is sent along-with API request. It is sent as a header with the API request. 
+
+`Authorization: Bearer <token>`
+
+The Signature is used to verify if the JWT was tampered with. Once the JWT is verified, the Payload can be decoded to get information about the request. 
+
+> Why is JWT used?
+
+- It helps communication to happen without authentication at every step by just verifying with JWT.
+- It is trustable as the JWT is digitally signed. 
+
+## Activity: Decode JWT
+
+Enough theory, isn't it? Let us do some activity. We will try to decode a JWT we saw above using an online decoder. You can use any decoder of your choice, or even better if you make your program. If you do make your program, do send a pull request. <br>
+
```

Comment:
`If you do make your program, do send a pull request`

This will give a nice opportunity to create a learner issue once the PR is merged. Just to give pointers :smiley:

Reply:
Will add issues after PR merge